platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/InvitationService.java

```diff
@@ -0,0 +1,103 @@
+package uno.cod.platform.server.core.service;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import uno.cod.platform.server.core.dto.challenge.ChallengeShowDto;
+import uno.cod.platform.server.core.dto.invitation.InvitationDto;
+import uno.cod.platform.server.core.dto.organization.OrganizationShowDto;
+import uno.cod.platform.server.core.dto.user.UserCreateDto;
+import uno.cod.platform.server.core.dto.user.UserShowDto;
+
+import javax.mail.MessagingException;
+import javax.mail.Transport;
+import javax.mail.internet.MimeMessage;
+import java.util.Arrays;
+
+@Service
+public class InvitationService {
+    private final UserService userService;
+    private final OrganizationService organizationService;
+    private final ChallengeService challengeService;
+    private final Environment environment;
+    private Logger log = Logger.getLogger(InvitationService.class.getName());
+
+    @Autowired
+    public InvitationService(UserService userService, OrganizationService organizationService, ChallengeService challengeService, Environment environment) {
+        this.userService = userService;
+        this.organizationService = organizationService;
+        this.challengeService = challengeService;
+        this.environment = environment;
+    }
+
+    public void invite(InvitationDto dto) {
+        OrganizationShowDto organization = organizationService.findById(dto.getOrganizationId());
+        if (organization == null) {
+            throw new IllegalArgumentException("invalid organization");
+        }
+        ChallengeShowDto challenge = challengeService.findById(dto.getChallengeId());
+        if (challenge == null) {
+            throw new IllegalArgumentException("invalid challenge");
+        }
+        UserShowDto user = userService.findByEmail(dto.getEmail());
+        if (user == null) {
+            UserCreateDto userCreateDto = new UserCreateDto();
+            userCreateDto.setEmail(dto.getEmail());
+            userCreateDto.setNick(dto.getEmail());
```

Comment:
nicknames can not contain "@" - so this will always fail
`@Pattern(regexp = "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$")` and https://github.com/coduno/platform/commit/48c8c11e7156fd2b9e3b0a0490a40718d3491cca